lib/engine.py

```diff
@@ -272,12 +268,12 @@ def convert_data_type(self, datatype):
             thispk = True
         elif thistype[0:3] == "ct-":
             thistype = thistype[3:]
-
         if thistype in list(self.datatypes.keys()):
             thistype = self.datatypes[thistype]
-
             if isinstance(thistype, tuple):
-                if len(datatype) > 1:
+                if datatype[0] == 'pk-auto':
+                    thistype =thistype
```

Comment:
What is the relationship between this conditional and the following conditional a few lines up:

```
         if thistype[0:3] == "pk-":
             thistype = thistype.lstrip("pk-")
             thispk = True
```

From a first read it seems like they're doing similar things maybe?